### PR TITLE
M3 app code

### PR DESCRIPTION
**Pull request by android-dev-lxl**
Add app code.

### CODE REVIEW

File: app/src/main/java/com/example/superheroes/MainActivity.kt [hunk 1]
```diff
@@ -38,8 +43,57 @@ class MainActivity : ComponentActivity() {
...
         }
     }
+
+    /**
+     * Composable that displays an app bar and a list of heroes.
+     */
+    @Composable
+    fun SuperheroesApp(modifier: Modifier = Modifier) {
+        Scaffold(
+            modifier = modifier.fillMaxSize(),
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/superheroes/MainActivity.kt:58]: pass in fillMaxSize() from the parent

File: app/src/main/java/com/example/superheroes/MainActivity.kt [hunk 2]
```diff
@@ -38,8 +43,57 @@ class MainActivity : ComponentActivity() {
...
                 }
             }
         }
     }
+
+    /**
+     * Composable that displays an app bar and a list of heroes.
+     */
+    @Composable
+    fun SuperheroesApp(modifier: Modifier = Modifier) {
```
Comment (MagicalMeghan) [inline app/src/main/java/com/example/superheroes/MainActivity.kt:56]: Dont need modifier passed in on main Composable

File: app/src/main/java/com/example/superheroes/MainActivity.kt (hunk 1 above)
Comment (MagicalMeghan) [inline app/src/main/java/com/example/superheroes/MainActivity.kt:58]: update this to Modifier.fillMaxSize()

File: app/src/main/java/com/example/superheroes/HeroesScreen.kt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                HeroListItem(
+                    hero = hero,
+                    modifier = Modifier
+                        .padding(horizontal = 16.dp, vertical = 8.dp)
+                        // Animate each list item to slide in vertically
+                        .animateEnterExit(
+                            enter = slideInVertically(
+                                animationSpec = spring(
+                                    stiffness = StiffnessVeryLow,
+                                    dampingRatio = DampingRatioLowBouncy
```
[the remark this answers is not in the document]
Comment (android-dev-lxl) [reply app/src/main/java/com/example/superheroes/HeroesScreen.kt:94]: Waiting for UX

File: app/src/main/java/com/example/superheroes/HeroesScreen.kt [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    }
+}
+
+@Composable
+fun HeroListItem(
+    hero: Hero,
+    modifier: Modifier = Modifier
+) {
+    Card(
+        elevation = CardDefaults.cardElevation(defaultElevation = 5.dp),
```
[the remark this answers is not in the document]
Comment (android-dev-lxl) [reply app/src/main/java/com/example/superheroes/HeroesScreen.kt:111]: Discussed this over chat

File: app/src/main/java/com/example/superheroes/HeroesScreen.kt [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+            // Start the animation immediately.
+            targetState = true
+        }
+    }
+
+    // Fade in entry animation for the entire list
+    AnimatedVisibility(
+        visibleState = visibleState,
+        enter = fadeIn(
+            animationSpec = spring(dampingRatio = DampingRatioLowBouncy)
```
[the remark this answers is not in the document]
Comment (MagicalMeghan) [reply app/src/main/java/com/example/superheroes/HeroesScreen.kt:78]: Still good to go to SME review!

File: app/src/main/java/com/example/superheroes/HeroesScreen.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (android-dev-lxl) [reply app/src/main/java/com/example/superheroes/HeroesScreen.kt:94]: Approved by Ivy

File: app/src/main/java/com/example/superheroes/HeroesScreen.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (android-dev-lxl) [reply app/src/main/java/com/example/superheroes/HeroesScreen.kt:78]: Approved by Ivy

**Review by yenerm — APPROVED** [review]

**Pull request merged**